cmd/internal/database/postgres/upgrade.go

```diff
@@ -133,7 +134,13 @@ func (db *Postgres) Upgrade(ctx context.Context) error {
 	}
 
 	// initdb -D /data/postgres-new
-	cmd := exec.Command(postgresInitDBCmd, "-D", newDataDirTemp)
+	// This is enabled by default since v18
+	initdbCommandArgs := []string{"-D", newDataDirTemp}
+	if oldBinaryVersionMajor == 17 {
+		initdbCommandArgs = append(initdbCommandArgs, "--no-data-checksums")
+	}
```

Comment:
if < 17 return fmt.Errorf("please update to 17 before upgrading to 18")

Reply:
done